deployments/extensions.go

```diff
@@ -0,0 +1,405 @@
+	}
+	defer os.RemoveAll(tmpDir)
+
+	manifestLocalPath, err := e.fetchFile(path, tmpDir)
+	if err != nil {
+		return errors.Wrap(err, "failed fetching file from "+path)
+	}
+
+	out, err := helpers.Kubectl(fmt.Sprintf("delete --filename %s --namespace %s", manifestLocalPath, ns))
+
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("kubectl delete failed:\n%s", out))
+	}
+	return nil
+}
+
+// TODO move under helpers
+func (e *Extension) installHelmPackage(name, chartPath, ns, valuesPath string) error {
+
+	tmpDir, err := ioutil.TempDir("", "fuseml-extension-chart")
+	if err != nil {
+		return errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	tarName := filepath.Base(chartPath)
+	if err = helpers.DownloadFile(chartPath, tarName, tmpDir); err != nil {
+		return errors.Wrap(err, "can't download helm chart for "+name)
+	}
+
+	chartLocalPath := filepath.Join(tmpDir, tarName)
+	valuesLocalPath := ""
+
+	if valuesPath != "" {
+		valuesLocalPath, err = e.fetchFile(valuesPath, tmpDir)
+		if err != nil {
+			return errors.Wrap(err, "failed fetching file from "+valuesPath)
+		}
+	}
+
+	helmCmd := fmt.Sprintf("helm install %s --create-namespace --values '%s' --namespace %s --wait %s", name, valuesLocalPath, ns, chartLocalPath)
+	currentdir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	if out, err := helpers.RunProc(helmCmd, currentdir, e.Debug); err != nil {
+		return errors.New(fmt.Sprintf("Failed installing %s chart: %s", name, out))
+	}
+
+	return nil
+}
+
+func (e *Extension) unInstallHelmPackage(ui *ui.UI, name, ns string) error {
+
+	currentdir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	out, err := helpers.WaitForCommandCompletion(ui, "Removing helm release "+name,
+		func() (string, error) {
+			helmCmd := fmt.Sprintf("helm uninstall '%s' --namespace '%s'", name, ns)
+			return helpers.RunProc(helmCmd, currentdir, e.Debug)
+		},
+	)
+	if err != nil {
+		if strings.Contains(out, "release: not found") {
+			ui.Exclamation().Msgf("%s helm release not found, skipping.\n", name)
+		} else {
+			return errors.Wrapf(err, "Failed uninstalling helm release %s: %s", name, out)
+		}
+	}
+
+	return nil
+}
+
+func deleteNamespace(c *kubernetes.Cluster, ui *ui.UI, ns string) error {
+
+	_, err := helpers.WaitForCommandCompletion(ui, "Deleting namespace "+ns,
+		func() (string, error) {
+			return "", c.DeleteNamespace(ns)
+		},
+	)
+	if err != nil {
+		return errors.Wrapf(err, "Failed deleting namespace %s", ns)
+	}
+	return nil
+}
+
+func (e *Extension) UnInstall(c *kubernetes.Cluster, ui *ui.UI, options *kubernetes.InstallationOptions) error {
+
+	namespace := e.desc.Namespace
+	if namespace == "" {
+		namespace = defaultNamespace
+	}
+	// based on installation type (script/helm/manifest), proceed with uninstallation of each install step
+	for _, step := range e.desc.Uninstall {
+
+		ns := step.Namespace
+		if ns == "" {
+			ns = namespace
+		}
+		switch step.Type {
```

Comment:
maybe consider adding a default, with some error for a unknown step.

Reply:
good point